lib/grpmat.gi

```diff
@@ -517,8 +517,29 @@ local   nice,img,module,b;
   return nice;
 end );
 
-InstallMethod( IsomorphismPermGroup,"matrix group", true,
-  [ IsMatrixGroup ], 10,
+
+#############################################################################
+##
+#M  IsomorphismPermGroup( <mat-grp> )
+##
+##  We want to use a method based on 'NicomorphismOfGeneralMatrixGroup'
+##  for the finite matrix group <mat-grp>.
+##
+##  If <mat-grp> does not know whether it is finite,
+##  there is currently no concurrent method,
+##  thus no rank shift is necessary in the first method installation.
+##
+##  If <mat-grp> knows to be finite, we want our method to beat the one for
+##  'IsGroup and IsFinite and IsHandledByNiceMonomorphism',
+##  so we install it with the same rank shift as that method.
+##  (Note that 'IsHandledByNiceMonomorphism' is implied by
+##  'IsMatrixGroup and IsFinite'.)
+##
+##  Installing just one method (with requirement 'IsMatrixGroup') would be
+##  difficult since we do not know the rank of 'IsGroup and IsFinite'.
```

Comment:
Isn't that what we have dynamic rank adjustment for? I.e., instead of a static rank adjustment like `10`, one can use a no-argument function which is used to recompute the rank adjustment each time we have to reorder methods due to new implications; e.g. `{} -> RankFilter(IsFinite and IsMatrixGroup)` is used in `lib/grpnice.gi:159`.

Reply:
Yes, one could use this feature.
Thus I should reformulate the sentence, because I do not think that this would be a good solution.

I think that the source of the problem is the current method with requirement `IsMatrixGroup`:
On the one hand, it has a low rank.
On the other hand, it shall be preferred to the method with requirement `IsGroup and IsHandledByNiceMonomorphism` in the situation that this is applicable, which typically happens when the argument is in `IsMatrixGroup and IsFinite`.

Upranking the `IsMatrixGroup` is not fair (compared to other methods) in the general case, and downranking the `IsGroup and IsHandledByNiceMonomorphism` method just because of the desired relation to the `IsMatrixGroup` method is also not fair.
A better solution would be to install the two methods without incremental ranks, and the method that is believed to be better in the concurrence situation should be installed explicitly for this situation.

(I did not dare to remove the extreme downranking of the `IsGroup and IsHandledByNiceMonomorphism` method. We can make sure that such a change does not mess up the code from the GAP distribution, but other GAP code might be affected.
Also, it is perhaps a mistake not to install the same method for the intersection situation `IsMatrixGroup and `IsHandledByNiceMonomorphism`. Should I change the pull request in this direction?)